operator/src/lib.rs

```diff
@@ -0,0 +1,323 @@
+                return Ok(ReconcileFunctionAction::Requeue(Duration::from_secs(10)));
+            }
+        };
+
+        self.zk_spec = Some(zk_spec);
+
+        Ok(ReconcileFunctionAction::Continue)
+    }
+
+    // TODO: Currently this only allows creation of Kafka Pods, it does not allow any updates or config changes
+    async fn reconcile(&mut self) -> KafkaReconcileResult {
+        let name = self.context.name();
+
+        let zk_spec = self.zk_spec.as_ref().ok_or_else(|| error::Error::ReconcileError("zk_spec missing, this is a programming error and should never happen. Please report in our issue tracker.".to_string()))?;
+
+        // This retrieves all the nodes from the referenced ZooKeeper cluster and creates the
+        // required connection string for Kafka
+        // TODO: Port is currently hardcoded, this needs to change and in general it might make sense to move this functionality to a ZooKeeper library
+        let zk_servers: String = zk_spec
+            .servers
+            .iter()
+            .map(|server| format!("{}:2181", server.node_name))
+            .collect::<Vec<String>>()
+            .join(",");
+
+        let mut labels = BTreeMap::new();
+        labels.insert("kafka-name".to_string(), name.clone());
+
+        let mut options = HashMap::new();
+        options.insert("zookeeper.connect".to_string(), zk_servers);
+
+        let mut handlebars = Handlebars::new();
+        handlebars
+            .register_template_string("conf", "{{#each options}}{{@key}}={{this}}\n{{/each}}")
+            .expect("template should work");
+
+        for broker in &self.kafka_spec.brokers {
+            trace!("Reconciling broker [{}]", broker.node_name);
+
+            // We first check whether we already have a Pod for this Node.
+            // If that's the case we'll use it and if not we create a new one.
+            let pod = match self
+                .existing_pods
+                .iter()
+                .find(|pod| podutils::is_pod_assigned_to_node(pod, &broker.node_name))
+            {
+                None => {
+                    info!(
+                        "Broker on node [{}] missing, creating now",
+                        broker.node_name
+                    );
+                    // TODO: These names need to contain a random component, also check the maximum length of names
+                    let pod_name = format!("kafka-{}-{}", name, broker.node_name);
+                    let cm_name = format!("kafka-{}", pod_name);
+
+                    // First we need to create/update the necessary pods...
+                    let pod = build_pod(
+                        &self.context.resource,
+                        &broker,
+                        &labels,
+                        &pod_name,
+                        &cm_name,
+                    )?;
+                    let pod = self.context.client.create(&pod).await?;
+
+                    // ...then we create the ConfigMap (one per pod):
```

Comment:
Nit: this comment and the next one sound very similar, I think this code creates the content for the map, no?